src/arduino.cc/builder/phases/libraries_builder.go

```diff
@@ -93,7 +93,7 @@ func fixLDFLAGforPrecompiledLibraries(ctx *types.Context, libraries []*types.Lib
 				name = strings.Replace(name, "lib", "", 1)
 				libs_cmd += "-l" + name + " "
 			}
-			ctx.BuildProperties[constants.BUILD_PROPERTIES_COMPILER_C_ELF_EXTRAFLAGS] += "\"-L" + path + "\" " + libs_cmd
+			ctx.BuildProperties[constants.BUILD_PROPERTIES_COMPILER_LDFLAGS] += "\"-L" + path + "\" " + libs_cmd
```

Comment:
Hi @facchinm,

I am attempting to use the 1.9.x beta of the Arduino IDE to try and use precompiled libs. I ran into an issue,
I faced 2 problems here
- No whitespace between the last existing value of compiler.ldflags and the appended string. (-L ..., -lmylib). Example: using the samd core 1.6.17, the concatenation looks like ```'--warn-section-align"-LC:\Users\```. Adding spaces after ```--warn-section-align``` simply gets removed out.
- These flags should be at the very end, typically after -lm etc. otherwise they won't be utilized by the linker appropriately. Basically the order matters. More info on [stackoverflow](https://stackoverflow.com/questions/5005363/undefined-reference-to-sin).